specs/phase1/beacon-chain.md

```diff
@@ -694,27 +729,30 @@ def apply_shard_transition(state: BeaconState, shard: Shard, transition: ShardTr
     )
     assert transition.start_slot == offset_slots[0]
 
-    # Reconstruct shard headers
     headers = []
     proposers = []
+    prev_gasprice = state.shard_states[shard].gasprice
     shard_parent_root = state.shard_states[shard].latest_block_root
     for i in range(len(offset_slots)):
-        if any(transition.shard_data_roots):
-            headers.append(ShardSignableHeader(
+        shard_block_length = transition.shard_block_lengths[i]
+        is_empty_proposal = shard_block_length == 0
+        shard_state = transition.shard_states[i]
+        if not is_empty_proposal:
+            proposal_index = get_shard_proposer_index(state, offset_slots[i], shard)
+            # Reconstruct shard headers
+            header = ShardBlockHeader(
                 shard_parent_root=shard_parent_root,
-                parent_hash=get_block_root_at_slot(state, get_previous_slot(state.slot)),
+                beacon_parent_root=get_block_root_at_slot(state, offset_slots[i]),
+                proposer_index=proposal_index,
                 slot=offset_slots[i],
                 body_root=transition.shard_data_roots[i]
-            ))
-            proposers.append(get_shard_proposer_index(state, shard, offset_slots[i]))
-            shard_parent_root = hash_tree_root(headers[-1])
+            )
+            shard_parent_root = hash_tree_root(header)
+            headers.append(header)
+            proposers.append(proposal_index)
 
-    # Verify correct calculation of gas prices and slots
-    prev_gasprice = state.shard_states[shard].gasprice
-    for i in range(len(offset_slots)):
-        shard_state = transition.shard_states[i]
-        block_length = transition.shard_block_lengths[i]
-        assert shard_state.gasprice == get_updated_gasprice(prev_gasprice, block_length)
+        # Verify correct calculation of gas prices and slots
+        assert shard_state.gasprice == compute_updated_gasprice(prev_gasprice, shard_block_length)
```

Comment:
I think gas prices and slots assertion can be moved below `shard_state = transition.shard_states[i]`

Optional comment, that's what I did in Prysm : )